src/vivarium_census_prl_synth_pop/components/population.py

```diff
@@ -274,12 +278,16 @@ def initialize_newborns(self, pop_data: SimulantData) -> None:
             additional_key="sex_of_child",
         )
         new_births["alive"] = "alive"
+        new_births["entrance_time"] = pop_data.creation_time
+        new_births["exit_time"] = pd.NaT
+
+        # Generate SSNs for newborns and prevent dups
         new_births["ssn"] = self.ssn_generator.generate(new_births).ssn
+        while len(new_births.ssn) != len(new_births.ssn.unique()) or sum(new_births.ssn.isin(ssns)) > 0:
+            new_births["ssn"] = self.ssn_generator.generate(new_births).ssn
```

Comment:
why does this need to happen for this line, but not for the original ssns created on population initialization?